lib/charms/opensearch/v0/opensearch_relation_provider.py

```diff
@@ -311,16 +333,22 @@ def create_opensearch_users(
                 role_name=username,
                 permissions=self.get_extra_user_role_permissions(extra_user_roles, index),
             )
-            roles = [username]
```

Comment:
We should keep this variable - I believe this was quite explanatory as to what `[username]` means exactly in the user creation workflow

Reply:
Solution fully re-worked: closing this thread.